presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1673,38 +1667,14 @@ private static void asyncRename(
             Path targetPath,
             List<String> fileNames)
     {
-        FileSystem fileSystem;
-        try {
-            fileSystem = hdfsEnvironment.getFileSystem(context, currentPath);
-        }
-        catch (IOException e) {
-            throw new PrestoException(HIVE_FILESYSTEM_ERROR, format("Error getting file system. Path: %s", currentPath), e);
-        }
-
+        FileSystem fileSystem = getFileSystem(hdfsEnvironment, context, currentPath);
         for (String fileName : fileNames) {
             Path source = new Path(currentPath, fileName);
             Path target = new Path(targetPath, fileName);
-            fileRenameFutures.add(executor.submit(() -> {
-                if (cancelled.get()) {
-                    return;
-                }
-                try {
-                    if (fileSystem.exists(target) || !fileSystem.rename(source, target)) {
-                        throw new PrestoException(HIVE_FILESYSTEM_ERROR, getRenameErrorMessage(source, target));
-                    }
-                }
-                catch (IOException e) {
-                    throw new PrestoException(HIVE_FILESYSTEM_ERROR, getRenameErrorMessage(source, target), e);
-                }
-            }));
+            fileRenameFutures.add(executor.submit(() -> renameFile(fileSystem, source, target, cancelled::get)));
```

Comment:
Can we dropped the `cancelled` parameter from `rename` method, and make this to be:
```java
fileRenameFutures.add(executor.submit(() -> {
   if (cancelled.get()) { return; }
    renameFile(fileSystem, source, target)));
}
```